sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/Command.scala

```diff
@@ -63,3 +63,14 @@ trait AnalysisOnlyCommand extends Command {
   // on the `AnalysisContext`
   def markAsAnalyzed(analysisContext: AnalysisContext): LogicalPlan
 }
+
+/**
+ * A logical node that does not expose its sub-nodes as children, but rather supervises them
+ * in an implementation-defined manner.
+ */
+trait SupervisingCommand extends LeafCommand {
+  /**
+   * Transform its supervised plan using `transformer` and returns a copy of `SupervisingCommand`
+   */
+  def withTransformedSupervisedPlan(transformer: (LogicalPlan) => LogicalPlan): LogicalPlan
```

Comment:
This can be rewritten as two methods:
```scala
def supervisedPlan: LogicalPlan
def withNewSupervisedPlan(new: LogicalPlan): LogicalPlan
```
But IMO the current impl is more concise